genesis/engine/scene.py

```diff
@@ -478,6 +487,43 @@ def add_light(
             mesh, color, intensity, morph.pos, morph.quat, revert_dir, double_sided, beam_angle
         )
 
+    @gs.assert_unbuilt
+    def add_light(
```

Comment:
Python does not support function overloading. You cannot rely on this approach to discriminate your two implementations for 'add_light'.